vault/external_tests/identity/login_mfa_totp_test.go

```diff
@@ -123,36 +124,40 @@ func TestLoginMfaGenerateTOTPTestAuditIncluded(t *testing.T) {
 	userpassPath := fmt.Sprintf("auth/userpass/login/%s", testuser1)
 
 	// MFA single-phase login
-	time.Sleep(5 * time.Second)
 	var secret *api.Secret
-	testhelpers.RetryUntil(t, 20*time.Second, func() error {
-		var err error
+	var err error
+	singlePhaseLoginFunc := func(methodIdentifier string) {
+		// this sleep is to let the TOTP refresh the code
+		time.Sleep(6 * time.Second)
```

Comment:
Rather than hard-code this, you could make it based on a variable that's also used in the `period` field for totpConfig.